content-security-policy/script-src/script-src-strict_dynamic_double_policy_honor_whitelist.sub.html

```diff
@@ -17,18 +17,18 @@
     <div id='log'></div>
 
     <script nonce='dummy'>
-        async_test(function(t) {
+        var cspViolationHandler;
+        test(function(t) {
             window.addEventListener('message', t.step_func(function(e) {
                 if (e.data === 'whitelisted-appendChild') {
                     t.done();
                 }
             }));
-            window.addEventListener('securitypolicyviolation', t.step_func(function(violation) {
-                if (violation.target.id !== 'whitelisted-appendChild') {
```

Comment:
Note that for URL-based violations, you can check `e.blockedURI` for these kinds of checks.